lib/pbench/server/auth/auth.py

```diff
@@ -11,170 +12,168 @@
-
-    @staticmethod
-    def set_logger(logger):
-        # Logger gets set at the time of auth module initialization
-        Auth.logger = logger
-
-    @staticmethod
-    def get_user_id() -> Optional[str]:
-        """
-        Returns the user id of the current authenticated user.
-        If the user not authenticated this would return None
-        """
-        user_id = None
-        user = Auth.token_auth.current_user()
-        if user:
-            user_id = str(user.id)
-        return user_id
-
-    def encode_auth_token(self, time_delta: datetime.timedelta, user_id: int) -> str:
-        """
-        Generates the Auth Token
-        Args:
-            time_delta: Token lifetime
-            user_id: Authorized user's internal ID
-        Returns:
-            JWT token string
-        """
-        current_utc = datetime.datetime.now(datetime.timezone.utc)
-        payload = {
-            "iat": current_utc,
-            "exp": current_utc + time_delta,
-            "sub": user_id,
-        }
-
-        # Get jwt key
-        jwt_key = self.get_secret_key()
-        return jwt.encode(payload, jwt_key, algorithm="HS256")
-
-    def get_secret_key(self):
-        try:
-            return os.getenv("SECRET_KEY", "my_precious")
-        except Exception as e:
-            Auth.logger.exception("Error {} getting JWT secret", e)
-
-    def get_auth_token(self, logger):
-        # get auth token
-        auth_header = request.headers.get("Authorization")
-
-        if not auth_header:
-            abort(
-                HTTPStatus.FORBIDDEN,
-                message="Please add authorization token as 'Authorization: Bearer <session_token>'",
-            )
-
-        try:
-            auth_schema, auth_token = auth_header.split()
-        except ValueError:
+token_auth = HTTPTokenAuth("Bearer")
+logger: Optional[Logger] = None
+
+
+def get_current_user_id() -> Optional[str]:
+    """
+    Returns the user id of the current authenticated user.
+    If the user not authenticated this would return None
+    """
+    user_id = None
+    user = token_auth.current_user()
+    if user:
+        user_id = str(user.id)
+    return user_id
+
+
+def encode_auth_token(time_delta: datetime.timedelta, user_id: int) -> str:
+    """
+    Generates the Auth Token
+    Args:
+        time_delta: Token lifetime
+        user_id: Authorized user's internal ID
+    Returns:
+        JWT token string
+    """
+    current_utc = datetime.datetime.now(datetime.timezone.utc)
+    payload = {
+        "iat": current_utc,
+        "exp": current_utc + time_delta,
+        "sub": user_id,
+    }
+
+    # Get jwt key
+    jwt_key = get_secret_key()
+    return jwt.encode(payload, jwt_key, algorithm="HS256")
+
+
+def get_secret_key() -> str:
+    """Returns JWT secret key."""
+    try:
+        return os.getenv("SECRET_KEY", "my_precious")
+    except Exception as e:
+        logger.exception("Error {} getting JWT secret", e)
```

Comment:
Since you are in here cleaning up, and if you come back in here for whatever reason, consider removing the redundant exception string formatted into the log message.

Reply:
Fixed.